training_config/biattentive_classification_network_elmo.json

```diff
@@ -0,0 +1,101 @@
+{
+  "dataset_reader":{
+    "type": "sst_tokens",
+    "use_subtrees": true,
+    "granularity": "5-class",
+    "token_indexers": {
+      "tokens": {
+        "type": "single_id"
+      },
+      "elmo": {
+        "type": "elmo_characters"
+      }
+    }
+  },
+  "validation_dataset_reader":{
+    "type": "sst_tokens",
+    "use_subtrees": false,
+    "granularity": "5-class",
+    "token_indexers": {
+      "tokens": {
+        "type": "single_id"
+      },
+      # If using ELMo in the BCN, add an elmo_characters
+      # token_indexers.
+      "elmo": {
+        "type": "elmo_characters"
+      }
+    }
+  },
+  "train_data_path": "https://s3-us-west-2.amazonaws.com/allennlp/datasets/sst/train.txt",
+  "validation_data_path": "https://s3-us-west-2.amazonaws.com/allennlp/datasets/sst/dev.txt",
+  "test_data_path": "https://s3-us-west-2.amazonaws.com/allennlp/datasets/sst/test.txt",
+  "model": {
+    "type": "bcn",
+    "text_field_embedder": {
+      "tokens": {
+        "pretrained_file": "https://s3-us-west-2.amazonaws.com/allennlp/datasets/glove/glove.840B.300d.txt.gz",
+        "type": "embedding",
+        "embedding_dim": 300,
+        "trainable": false
+      }
+    },
+    # BCN configs that use ELMo should not have an elmo_token_embedder
```

Comment:
Can you put this comment above the text_field_embedder and say _why_ not, rather than just stating it. Maybe "The BCN model will consume the arrays generated by the ELMo token_indexer independently of the text_field_embedder, so we miss out the key for that here".

Reply:
good idea, thanks!